### ISSUE TITLE
Issue with Generic function and Literal type

### ISSUE THREAD

**Issue by bloussou** [microsoft/pyright#1643]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Generic function are not working for type `Literal` because type `str` is returned.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import Literal, TypeVar

_T = TypeVar("_T")

def get_or_fallback(arg: _T) -> _T:
    return arg

value: Literal["test"] = get_or_fallback("test") # Expression of type "str" cannot be assigned to declared type "Literal['test']"
```

**VS Code extension or command-line**
This behavior is present since 118.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.123, which I just published. It will also be included in the next release of pylance.
